### ISSUE TITLE
Suggestion: More graceful error handling

### ISSUE THREAD

**Issue by btemperton** [shafferm/DRAM#33]
Currently, if DRAM hits a problem finding a key in either the gtdbtk or checkm files, it crashes, e.g.:

```
Traceback (most recent call last):
  File "<environment>/DRAM/bin/DRAM.py", line 145, in <module>
    args.func(**args_dict)
  File "<environment>/DRAM/lib/python3.8/site-packages/mag_annotator/annotate_bins.py", line 963, in annotate_bins_cmd
    annotate_bins(fasta_locs, output_dir, min_contig_size, prodigal_mode, trans_table, bit_score_threshold,
  File "<environment>/DRAM/lib/python3.8/site-packages/mag_annotator/annotate_bins.py", line 1007, in annotate_bins
    all_annotations['bin_taxonomy'] = [gtdb_taxonomy.loc[i, 'classification'] for i in all_annotations.fasta]
  File "<environment>/DRAM/lib/python3.8/site-packages/mag_annotator/annotate_bins.py", line 1007, in <listcomp>
    all_annotations['bin_taxonomy'] = [gtdb_taxonomy.loc[i, 'classification'] for i in all_annotations.fasta]
  File "<environment>/DRAM/lib/python3.8/site-packages/pandas/core/indexing.py", line 873, in __getitem__
    return self._getitem_tuple(key)
  File "<environment>/DRAM/lib/python3.8/site-packages/pandas/core/indexing.py", line 1044, in _getitem_tuple
    return self._getitem_lowerdim(tup)
  File "<environment>/DRAM/lib/python3.8/site-packages/pandas/core/indexing.py", line 786, in _getitem_lowerdim
    section = self._getitem_axis(key, axis=i)
  File "<environment>/DRAM/lib/python3.8/site-packages/pandas/core/indexing.py", line 1110, in _getitem_axis
    return self._get_label(key, axis=axis)
  File "<environment>/DRAM/lib/python3.8/site-packages/pandas/core/indexing.py", line 1059, in _get_label
    return self.obj.xs(label, axis=axis)
  File "<environment>/DRAM/lib/python3.8/site-packages/pandas/core/generic.py", line 3488, in xs
    loc = self.index.get_loc(key)
  File "<environment>/DRAM/lib/python3.8/site-packages/pandas/core/indexes/base.py", line 2893, in get_loc
    raise KeyError(key) from err
KeyError: 'HiC_MES_bin_105'

```
This can happen days into a runtime after successful annotation of many bins. The crash prevents the `annotations.tsv` and other downstream outputs to be created, which means the whole thing needs to be started again. It would be better if upon failure, the bin responsible was either removed from the output or flagged in some way, allowing the pipeline to continue.

In short, unless errors are fatal to the pipeline, they need to be handled in a way that doesn't compromise successfully completed steps.

**Comment by shafferm** [comment]
This is a great suggestion and was definitely on our list. Thanks to your push I have pushed a quick fix for this to the master branch. It currently will replace missing taxonomy with the fasta/bin name, the completeness with 0 and the contamination with 100 as well as issue a warning for each one missing. I'll try to figure out a more graceful way to handle it in an upcoming release.

If you want to try out the fix you can install from the master branch. You can save your config file, update your DRAM installation from the master branch of this repo and then reimport your config file. Instructions on how to do this are here: https://github.com/shafferm/DRAM/wiki/5.-Managing,-updating-or-moving-a-DRAM-installation-and-databases#updating-your-dram-installation. Otherwise I will update this issue when a new version with the fix has been released. I'm planning a new release for the second half of this week or early next week so should be fixed soon.

**Comment by shafferm** [comment]
This has been implemented and pushed to the master branch. Will be included in the next release.
